frontend/src/shape/BarStatShape.ts

```diff
@@ -0,0 +1,15 @@
+import { assertEqual, Shape } from "./Shape";
+import { StatMold, StatShape } from "./StatShape";
+
+export type BarStatShape = StatShape & {
+    type: "bar";
+};
+
+export class BarStatMold extends Shape<BarStatShape> {
+    statMold = new StatMold();
+
+    protected override throwIfMisfit(obj: any) {
+        this.statMold.assert(obj);
+        assertEqual(obj["type"], "bar");
+    }
+}
```

Comment:
I like this, we spoke about separation of Shape and Mold I know you are doing that. I like the Shape to Mold names and categorization.